### DISCUSSION TITLE
lightgallery 两个问题

### DISCUSSION THREAD

**Discussion by HonorKnight8** [HEIGE-PCloud/DoIt#128] [Q&A] [answered]
问题1：图片点开了就是一张图片的页面，没找到要怎么开启画廊效果

参考了这位大佬的说法，并没有解决
https://lewky.cn/posts/hugo-4.html/#lightgallery%E5%9B%BE%E7%89%87%E7%9B%B8%E5%86%8C%E5%8A%9F%E8%83%BD%E6%97%A0%E6%B3%95%E4%BD%BF%E7%94%A8

问题2：就是第一次访问时图片尺寸正常，刷新，或者点开图片再回退之后，图片尺寸变得非常小
说是lazysizes的问题
https://lewky.cn/posts/hugo-4.html/#lightgallery%E5%90%AF%E7%94%A8%E6%97%B6%E7%AC%AC%E4%BA%8C%E6%AC%A1%E5%8A%A0%E8%BD%BD%E9%A1%B5%E9%9D%A2%E6%97%B6%E5%9B%BE%E7%89%87%E4%BC%9A%E5%8F%98%E5%B0%8F
https://github.com/dillonzq/LoveIt/issues/311
https://github.com/aFarkas/lazysizes/issues/508
除了用css把图片全部设置成100%宽，还有好的方法吗？有的图片是纵向的。100%宽太影响整个页面了

**Comment by HEIGE-PCloud** [comment]
1. 使用 [image shortcode](https://hugodoit.pages.dev/zh-cn/theme-documentation-extended-shortcodes/#image) 来启用 lightgallery
2. 请问这个问题可以在这两个页面上复现吗？
    - https://hugodoit.pages.dev/extend-shortcodes-tests/#image
    - https://hugodoit.pages.dev/markdown-tests/#images

**Reply by HonorKnight8** [reply]
奇怪，为什么大佬这个这么正常

**Reply by HonorKnight8** [reply]
我是不是哪里设置不对
主题自带的exampleSite，要怎么搭起来？我试试对照着改改

**Reply by HEIGE-PCloud** [reply]
在 DoIt 文件夹下执行 `hugo server --source=exampleSite` 来运行 exampleSite

**Comment by HonorKnight8** [comment, marked as answer]
https://octodex.github.com/images/stormtroopocat.jpg 
问题一：像这个效果

**Reply by HEIGE-PCloud** [reply]
是的，`![Alt Text](image)`的语法不会启用 lightgallery。请用 [image shortcode](https://hugodoit.pages.dev/zh-cn/theme-documentation-extended-shortcodes/#image) 来使用 lightgallery。

**Reply by HonorKnight8** [reply]
我是用的shortcode

大佬帮我看下
http://helloworld.monster/posts/a_maddening_issue_about_dependency_in_idea/
http://helloworld.monster/posts/a_maddening_issue_about_dependency_in_idea/index.md

**Reply by HEIGE-PCloud** [reply]
在 frontmatter 里需要开启 `lightgallery: true`

相关文档 https://hugodoit.pages.dev/zh-cn/theme-documentation-content/#front-matter

**Reply by HEIGE-PCloud** [reply]
对于你的站点，还有以下问题

1. 在 `config.toml` 中需要设置 `hasCJKLanguage = true` 来保证生成的 Markdown 源文件不是乱码。
2. 在目录设置中需要配置正确的层级来保证右侧的 TOC 正确生成。
```toml
  [markup.tableOfContents]
    startLevel = 2
    endLevel = 6
```
相关文档 https://hugodoit.pages.dev/zh-cn/theme-documentation-basics/

**Reply by HonorKnight8** [reply]
好，谢谢大佬
可能我配置文件到处乱抄的，顺序有问题，我参照 exampleSite 再改改

**Comment by HonorKnight8** [comment]
找到原因了，我之前不知道哪里抄的配置
在 frontmatter 里靠后的位置插了一条 lightgallery: false
然后早上检查不细心

frontmatter里是这样：
...
lightgallery: true
...
lightgallery: false
...

嗯，又是一个应该面壁的问题

**Reply by HonorKnight8** [reply]
这样，一下两个问题都没了

**Reply by HEIGE-PCloud** [reply]
好耶！

**Reply by HonorKnight8** [reply]
顺便跟着 exampleSite 整理了一下目录和配置文件
